python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1480,6 +1491,73 @@ def latest_materialization_event(
             AssetKey.from_coercible(key)
         )
 
+    ##### Partition methods
+
+    @public
+    @property
+    def has_partition_key(self) -> bool:  # TODO - maybe this one can be replaced by something else?
```

Comment:
from my understanding, this method is mostly used to check if a run is partitioned or not. In that case, I think a name like `is_partitioned` is better